molo/commenting/tests/test_models.py

```diff
@@ -125,3 +126,70 @@ def test_auto_remove_for_non_remove_flag(self):
             )
         altered_comment = MoloComment.objects.get(pk=comment.pk)
         self.assertFalse(altered_comment.is_removed)
+
+
+class CommentingSettingsTest(TestCase, MoloTestCaseMixin):
+    """Test if the anonymous commengting alias value is set properly."""
+
+    def setUp(self):
+        self.mk_main()
+        self.main = Main.objects.all().first()
+        self.language_setting = Languages.objects.create(
+            site_id=self.main.get_site().pk)
+        self.english = SiteLanguageRelation.objects.create(
+            language_setting=self.language_setting,
+            locale='en',
+            is_active=True)
+        self.user = User.objects.create_user(
+            'test', 'test@example.org', 'test')
+        self.content_type = ContentType.objects.get_for_model(self.user)
+        self.yourmind = self.mk_section(
+            self.section_index, title='Your mind')
+        self.article1 = self.mk_article(
+            title='article 1', slug='article-2', parent=self.yourmind)
+        self.setting = CommentingSettings.for_site(self.site)
+
+    def test_get_comments_anonymous(self):
+        article = self.article1
+        MoloComment.objects.create(
+            content_object=article, object_pk=article.id,
+            content_type=ContentType.objects.get_for_model(article),
+            site=Site.objects.get_current(), user=self.user,
+            comment='This is a comment, you dig?', submit_date=timezone.now())
+        response = self.client.get(
+            reverse('molo.commenting:more-comments', args=(article.pk,)))
+        self.assertContains(response, "Anonymous")
+
+    def test_get_comments_anonymous_alias(self):
+        self.setting.commenting_anonymous = "Little Sister"
+        self.setting.save()
+        article = self.article1
+        MoloComment.objects.create(
+            content_object=article, object_pk=article.id,
+            content_type=ContentType.objects.get_for_model(article),
+            site=Site.objects.get_current(), user=self.user,
+            comment='This is another comment', submit_date=timezone.now())
+        response = self.client.get(
+            reverse('molo.commenting:more-comments', args=(article.pk,)))
+        self.assertContains(response, "Little Sister")
+
+    def test_anonymous_comment_translation(self):
```

Comment:
This test creates a comment on the english article, could we make a comment on the french article and test that the comment on the french article contains the french word for anonymous when shown on the front end?

Reply:
@codiebeulaine I'm not sure how the translations work but apparently the word won't be displayed because it won't have been translated in Pootle or something? So Sewa can't test that it gets translated. If you know of some way to do this or if there's somewhere else it's done that we could look at that would be great.

Reply:
@KaitCrawford anonymous should be translated in the PO files. It was using anonymous before, so it should be translated. Just need to make sure that the french translations or any other translations will still pull through. So if you get the article url as a response, and assert that the response contains the french version of anonymous, it should be there